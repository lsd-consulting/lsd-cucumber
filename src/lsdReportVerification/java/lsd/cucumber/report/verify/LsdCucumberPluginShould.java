package lsd.cucumber.report.verify;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

import static java.nio.file.Paths.get;
import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;

class LsdCucumberPluginShould {

    @Test
    void generateMenu() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/lsd-index.html"));
    }

    @Test
    void generateReportForBasicScenario() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Scenario.html"));
    }

    @Test
    void generateReportForScenarioWithDataTable() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Data_table.html"));
    }

    @Test
    void generateForScenarioOutline() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Scenario_outline.html"));
    }

    private File copyOfFile(String path) throws IOException {
        Path original = get(path);
        Path copy = Path.of(original.getParent().toString(), "copy_" + original.getFileName());
        Files.copy(original, copy, REPLACE_EXISTING);
        return copy.toFile();
    }
}
