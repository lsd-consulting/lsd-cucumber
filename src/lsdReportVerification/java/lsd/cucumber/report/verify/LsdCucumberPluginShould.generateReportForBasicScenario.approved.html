<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns="http://www.w3.org/1999/html" xml:lang="en" lang="en">
<head>
    <title>scenario</title>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/lsd-consulting/lsd-core@7/src/main/resources/static/style.min.css">

</head>
<body>
<main role="main">
    
    <h1 class="logo">LSD</h1>
    <h1>scenario</h1><article id="6">
        <span class="articleStart"></span>
        <h2 class="success">Basic scenario</h2><section class="description success">
            <h3 class="success">Description</h3>
            Given  given1</br>When  when1</br>Then  then1
        </section>
        
        
        <section class="sequence diagram svg success">
            <h3 class="success">Sequence Diagram</h3>
            <figure><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" data-diagram-type="SEQUENCE" height="197px" preserveAspectRatio="none" style="width:207px;height:197px;background:#FFFFFF;" version="1.1" viewBox="0 0 207 197" width="207px" zoomAndPan="magnify"><title>Given given1</title><defs/><g><text fill="#000000" font-family="Hack" font-size="22" font-weight="bold" lengthAdjust="spacing" textLength="147.7695" x="36.2275" y="36.2695">Given given1</text><g><title>Source</title><rect fill="#000000" fill-opacity="0.00000" height="78.6211" width="8" x="45.7637" y="83.3984"/><line style="stroke:#000000;stroke-width:1;" x1="49" x2="49" y1="83.3984" y2="162.0195"/></g><g><title>Destination</title><rect fill="#000000" fill-opacity="0.00000" height="78.6211" width="8" x="151.376" y="83.3984"/><line style="stroke:#000000;stroke-width:1;" x1="154.5273" x2="154.5273" y1="83.3984" y2="162.0195"/></g><g class="participant participant-head" data-participant="Source"><rect fill="#FFFFFF" height="30.4883" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1;" width="59.5273" x="20" y="51.9102"/><text fill="#000000" font-family="Hack" font-size="14" lengthAdjust="spacing" textLength="45.5273" x="27" y="72.4453">Source</text></g><g class="participant participant-tail" data-participant="Source"><rect fill="#FFFFFF" height="30.4883" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1;" width="59.5273" x="20" y="161.0195"/><text fill="#000000" font-family="Hack" font-size="14" lengthAdjust="spacing" textLength="45.5273" x="27" y="181.5547">Source</text></g><g class="participant participant-head" data-participant="Destination"><rect fill="#FFFFFF" height="30.4883" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1;" width="91.6973" x="109.5273" y="51.9102"/><text fill="#000000" font-family="Hack" font-size="14" lengthAdjust="spacing" textLength="77.6973" x="116.5273" y="72.4453">Destination</text></g><g class="participant participant-tail" data-participant="Destination"><rect fill="#FFFFFF" height="30.4883" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1;" width="91.6973" x="109.5273" y="161.0195"/><text fill="#000000" font-family="Hack" font-size="14" lengthAdjust="spacing" textLength="77.6973" x="116.5273" y="181.5547">Destination</text></g><g class="message" data-participant-1="Source" data-participant-2="Destination"><polygon fill="#000000" points="143.376,110.709,153.376,114.709,143.376,118.709,147.376,114.709" style="stroke:#000000;stroke-width:1;"/><line style="stroke:#000000;stroke-width:1;" x1="49.7637" x2="149.376" y1="114.709" y2="114.709"/><a href="#id1" target="_self" title="label1" xlink:actuate="onRequest" xlink:href="#id1" xlink:show="new" xlink:title="label1" xlink:type="simple"><text fill="#333333" font-family="Hack" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="38.3398" x="56.7637" y="109.9668">label1</text></a></g><g class="message" data-participant-1="Destination" data-participant-2="Source"><polygon fill="#000000" points="60.7637,140.0195,50.7637,144.0195,60.7637,148.0195,56.7637,144.0195" style="stroke:#000000;stroke-width:1;"/><line style="stroke:#000000;stroke-width:1;" x1="54.7637" x2="154.376" y1="144.0195" y2="144.0195"/><a href="#id2" target="_self" title="label2" xlink:actuate="onRequest" xlink:href="#id2" xlink:show="new" xlink:title="label2" xlink:type="simple"><text fill="#333333" font-family="Hack" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="38.3398" x="66.7637" y="139.2773">label2</text></a></g></g></svg></figure>
            <div onclick="location.href='#!';" id="4" class="overlay">
                <div class="popup" onclick="event.stopPropagation();">
                    <h2>PlantUml markup</h2>
                    <span class="copy" title="copy" onclick="copyToClipboard('sequence_uml_code4')"></span>
                    <a class="close" href="#!">&times;</a>
                    <div class="content">
                        <pre><code id="sequence_uml_code4">@startuml

!theme mono
skinparam svgLinkTarget _self

hide unlinked



participant Source
participant Destination



title Given given1

Source -&gt; Destination: [[#id1 {label1} label1]]
Destination -&gt; Source: [[#id2 {label2} label2]]


@enduml

</code></pre>
                    </div>
                </div>
            </div>
            <sub><a href="#4">[source]</a></sub>
            <sub><a href="#5">[component diagram]</a></sub>
        </section>
        <!-- Popup for component diagram -->
        <div onclick="location.href='#!';" id="5" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>Component Diagram</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" data-diagram-type="DESCRIPTION" height="153px" preserveAspectRatio="none" style="width:136px;height:153px;" version="1.1" viewBox="0 0 136 153" width="136px" zoomAndPan="magnify"><defs/><g><!--entity Source--><g class="entity" data-entity="Source" data-source-line="5" data-uid="ent0002" id="entity_Source"><rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1;" width="88.5352" x="26.5" y="17"/><rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1;" width="15" x="95.0352" y="22"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1;" width="4" x="93.0352" y="24"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1;" width="4" x="93.0352" y="28"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48.5352" x="41.5" y="51.0752">Source</text></g><!--entity Destination--><g class="entity" data-entity="Destination" data-source-line="6" data-uid="ent0003" id="entity_Destination"><rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1;" width="119.7617" x="11" y="100"/><rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1;" width="15" x="110.7617" y="105"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1;" width="4" x="108.7617" y="107"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1;" width="4" x="108.7617" y="111"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="79.7617" x="26" y="134.0752">Destination</text></g><g class="link" data-entity-1="Source" data-entity-2="Destination" data-uid="lnk4" id="link_Source_Destination"><polygon fill="#000000" points="64.8648,99.7771,68.1917,90.5071,64.4968,94.7906,60.2134,91.0958,64.8648,99.7771" style="stroke:#000000;stroke-width:1;"/><path d="M64.8766,64.063 C64.0451,75.204 63.5996,82.6307 64.4233,93.7933" fill="none" style="stroke:#000000;stroke-width:1;"/></g><g class="link" data-entity-1="Destination" data-entity-2="Source" data-uid="lnk5" id="link_Destination_Source"><polygon fill="#000000" points="77.1234,64.063,73.8043,73.3358,77.4955,69.0491,81.7821,72.7403,77.1234,64.063" style="stroke:#000000;stroke-width:1;"/><path d="M77.1352,99.7771 C77.9589,88.6144 78.4015,81.1874 77.5699,70.0464" fill="none" style="stroke:#000000;stroke-width:1;"/></g></g></svg></pre>
                </div>
                <sub><a href="#uml5">[source]</a></sub>
            </div>
        </div>
        <!-- Popup for component diagram UML via component diagram popup -->
        <div onclick="location.href='#!';" id="uml5" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>PlantUml markup</h2>
                <span class="copy" title="copy" onclick="copyToClipboard('component_uml_code5')"></span>
                <a class="close" href="#5">&times;</a><!--show previous popup on close-->
                <div class="content">
                    <pre><code id="component_uml_code5">@startuml
!pragma layout smetana
!theme plain
skinparam backgroundColor transparent

component Source
component Destination


[Source] --&gt; Destination
[Destination] --&gt; Source


remove @unlinked
@enduml</code></pre>
                </div>
            </div>
        </div>
        <!-- Section containing data for sequence diagram popups --><div onclick="location.href='#!';" id="id1" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label1</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div><div onclick="location.href='#!';" id="id2" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label2</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div>
    </article>
</main>
</body>
<footer>
    <script src="https://cdn.jsdelivr.net/npm/d3@7"></script>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11/build/styles/dark.min.css">
    <script src="https://cdn.jsdelivr.net/gh/lsd-consulting/lsd-core@7/src/main/resources/static/custom.min.js"></script>

    <script>
        addSliderForSvgZoom();
        highlightLifelinesWhenClicked();
    </script>
    <script src="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11/build/highlight.min.js"></script>
    <script>hljs.highlightAll();</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js"></script>
    <script>
        highlightKeywords();
        unMarkSvg();
    </script>
</footer>
</html>
